app/api/user_api/v1/sessions.rb

```diff
@@ -43,19 +45,23 @@ def create_access_token(expires_in:, account:, application:)
           end
 
           unless account.otp_enabled
+            account.update!(failed_attempts: 0)
             return create_access_token expires_in: declared_params[:expires_in],
                                        account: account,
                                        application: application
           end
 
           if declared_params[:otp_code].blank?
+            account.update!(failed_attempts: account.failed_attempts += 1)
             error!('The account has enabled 2FA but OTP code is missing', 403)
           end
 
           unless Vault::TOTP.validate?(account.uid, declared_params[:otp_code])
+            account.update(failed_attempts: account.failed_attempts += 1)
```

Comment:
I suggest to add method to `account` because you have 3 duplication of that update